addon/utils/url-builder.js

```diff
@@ -1,8 +1,17 @@
-import param from 'jquery-param';
+import URL from 'url';
+import { isEmpty } from '@ember/utils';
 
 export default function(url, path, queryParams) {
-  let query = param(queryParams);
-  let pathUrl = url.charAt(url.length - 1) === '/' ? `${url}${path}` : `${url}/${path}`;
-
-  return query ? `${pathUrl}?${query}` : pathUrl;
+  let urlObj = new URL(url, window.location.origin);
```

Comment:
This will not work in fastboot :/ There is no window object

Reply:
@GreatWizard Since we don't care about the host name here, I'd recommend using a dummy hostname.

```js
let urlObj = new URL(url, 'https://example.com');
```